Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -0,0 +1,190 @@
+    func fetch(dependenciesDirectory: AbsolutePath, dependencies: SwiftPackageManagerDependencies) throws
+}
+
+// MARK: - Swift Package Manager Interactor
+
+public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
+    ) {
+        self.fileHandler = fileHandler
+    }
+
+    public func fetch(dependenciesDirectory: AbsolutePath, dependencies: SwiftPackageManagerDependencies) throws {
+        logger.info("We are starting to fetch the Swift Package Manager dependencies.", metadata: .section)
+
+        try fileHandler.inTemporaryDirectory { temporaryDirectoryPath in
+            // prepare paths
+            let pathsProvider = SwiftPackageManagerPathsProvider(
+                dependenciesDirectory: dependenciesDirectory,
+                temporaryDirectoryPath: temporaryDirectoryPath
+            )
+
+            // prepare for installation
+            try prepareForInstallation(pathsProvider: pathsProvider, dependencies: dependencies)
+
+            // build command
+            let command = buildCommand(packagePath: temporaryDirectoryPath)
+
+            // run `Swift Package Manager`
+            logger.info("SwiftPackageManager:", metadata: .subsection)
+            try System.shared.runAndPrint(command)
+
+            // post installation
+            try postInstallationActions(pathsProvider: pathsProvider)
+        }
+
+        logger.info("Swift Package Manage dependencies were fetched successfully.", metadata: .subsection)
+    }
+
+    // MARK: - Installation
+
+    private func prepareForInstallation(pathsProvider: SwiftPackageManagerPathsProvider, dependencies: SwiftPackageManagerDependencies) throws {
+        // copy `.build` directory from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationSwiftPackageManagerBuildDirectory) {
+            try copyDirectory(
+                from: pathsProvider.destinationSwiftPackageManagerBuildDirectory,
+                to: pathsProvider.temporarySwiftPackageManagerBuildDirectory
+            )
+        }
+
+        // copy `Package.resolved` directory from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationPackageResolvedPath) {
+            try copyDirectory(
+                from: pathsProvider.destinationPackageResolvedPath,
+                to: pathsProvider.temporaryPackageResolvedPath
+            )
+        }
+
+        // create `Package.swift`
+        let packageManifestContent = dependencies.stringValue()
+        let packageManifestPath = pathsProvider.temporaryDirectoryPath.appending(component: "Package.swift")
+        try fileHandler.write(packageManifestContent, path: packageManifestPath, atomically: true)
+
+        // log
+        logger.info("Package.swift:", metadata: .subsection)
```

Comment:
This content is too verbose for the user so I'd rather use `logger.debug`

Reply:
I changed it in `CarthageInteractor` as well.